### DISCUSSION TITLE
Run a single-line command from GDL in a shell script

### DISCUSSION THREAD

**Discussion by claudioPaga** [gnudatalanguage/gdl#1212] [Q&A] [answered]
Hi,

is there an equivalent way with GDL to run a single-line command in a shell script like this

idl -e "test_example"

I've tried naively

gdl -e "test_example"

but with no success. Thanks

**Comment by alaingdl** [comment, marked as answer]
Not sure to understand you ...

I can do :
```
~/GDL/gdl-1.0.1-git211111CMake$ ./src/gdl -quiet -e "print, 10*20"
     200
```
which run exactly as IDL ...

and you can access to in-line doc : `gdl -h`

HTH

A.

**Reply by claudioPaga** [reply]
Thanks for the immediate reply and apologies for not being clear.

What I've meant was the ability to run a program like that, so in my example

test_example.pro 

is an IDL/GDL program and I run it from the shell command line using

idl -e "test_example"

Thanks again, Claudio

**Reply by GillesDuvert** [reply]
yes, gdl -e "something" . This is largely used for years with GDL.
provided "something" is ok with GDL.
example `gdl -e "a=dist(10) & print,a"`

**Reply by claudioPaga** [reply]
thanks Gilles

**Comment by alaingdl** [comment]
I created a small procedure maned `claudio.pro` in current directory
``` 
$ more claudio.pro 
pro claudio
print, 10*20
plot, findgen(10)
wait, 4
end
```
then run it without problem with : `gdl -e claudio` or `gdl -e "claudio"`, as I would have done with IDL (and yes I tested that too :smile: ) 

Seems to be mandatory that the radical of the file and the name of the procedure are the sames.

**Reply by claudioPaga** [reply]
Thank you Alain for your answer and your time, your example works perfectly.

I've been trying to install and use GDL in a Docker image. GLD installed fine, but when I tried to test it running a simple test script  I was getting an error

root@a6463e2cfbc0:/# gdl -quiet -e "plot_test"
% WARNING: your version of the GraphicsMagick library will truncate images to 16 bits per pixel
% Compiled module: PLOT_TEST.
% PRINT: Variable is undefined: PLOT_TEST
% Execution halted at: PLOT_TEST              /plot_test.pro

But this is evidently not an issue with using gdl -e but something different, as when I try it outside Docker it works fine. I'll have to look into this, thanks again,
Claudio

**Reply by alaingdl** [reply]
may-be just a problem of path ? what is "/plot_test.pro". Why no just "/plot_test.pro"

When you will have a solution, please report, especially if it is a trick related to docker

**Reply by GillesDuvert** [reply]
GDL **IS** used in docker images since long.
[This facility](http://www.jmmc.fr/english/tools/data-analysis/oimaging/) is a java app that dialogs with a server on which several image-reconstruction programs run. All these programs run in docker. one of the, WISARD, is a GDL procedure calling GDL (so, in the docker) , with `gdl -e 'wisard, input="this", output="that"'` .

**Comment by claudioPaga** [comment]
I now have things set up as I needed in the Docker container, with GDL installed along with the Astronomy and Coyote Libray and all  seems to be running fine.

In case it might be useful to others, I copy here the content of the Dockerfile I've used to create the Docker image. Thanks again to all for the replies and feedback.

#Step 1
#Dockerfile to build an image
#Pull ubuntu 20.04 base image
FROM ubuntu:20.04
MAINTAINER SOC

ENV DEBIAN_FRONTEND=noninteractive

#Step 2
#Essential tools and xvfb
RUN apt-get update && apt-get install -y \\
    software-properties-common \\
    g++ \\
    cmake \\
    git \\
    sudo \\
    unzip \\
    curl \\
    wget \\
    bzip2 \\
    net-tools \\
    xterm \\
    traceroute \\
    dnsutils \\
    iputils-ping \\
    vim \\
    xvfb

#Step 3
#GDL installation
RUN mkdir GDL && cd GDL && \\
    git clone https://github.com/gnudatalanguage/gdl && \\
    cd gdl && \\
    ./scripts/build_gdl.sh prep && \\
    ./scripts/build_gdl.sh configure && \\
    ./scripts/build_gdl.sh build && \\
    cd .. && \\
    wget https://idlastro.gsfc.nasa.gov/ftp/astron.tar.gz && \\
    tar zxvf astron.tar.gz && \\
    wget https://idlastro.gsfc.nasa.gov/ftp/coyote_astron.tar.gz && \\
    tar zxvf coyote_astron.tar.gz

ENV GDL_PATH="+/GDL/pro:+/GDL/gdl/src/pro"
RUN  /bin/echo alias gdl=/GDL/gdl/build/src/gdl >> /root/.bashrc

#Step 4
#Shell script
COPY . ./

#In case graphics are required
ENV DISPLAY=host.docker.internal:0

**Comment by GillesDuvert** [comment]
thanks, there is now a new chapter 'GDL in a Docker container' on our wiki :+1:
